api/envoy/config/wasm/v2alpha/wasm.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+
+import "google/protobuf/any.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.wasm.v2alpha";
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package = "envoy.extensions.wasm.v3alpha";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+// [#not-implemented-hide:] pending implementation.
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or the name of the registered Null
+  // VM plugin) to determine which VM will be used for the plugin. All plugins which use the same
+  // *vm_id* and code will use the same VM. May be left blank. Sharing a VM between plugins can
+  // reduce memory utilization and make sharing of data easier which may have security implications.
+  // See ref: "TODO: add ref" for details.
+  string vm_id = 1;
+
+  // The Wasm runtime type (either "v8" or "null" for code compiled into Envoy).
+  string runtime = 2 [(validate.rules).string = {min_bytes: 1}];
+
+  // The Wasm code that Envoy will execute.
+  api.v2.core.AsyncDataSource code = 3;
+
+  // The Wasm configuration used in initialization of a new VM (proxy_on_start).
+  google.protobuf.Any configuration = 4;
+
+  // Allow the wasm file to include pre-compiled code on VMs which support it.
+  bool allow_precompiled = 5;
```

Comment:
Could you add a big warning that this should be enabled only for modules from trusted sources? Thanks!

Reply:
Done.

Reply:
Sorry, I should have been more explicit. I meant the `.. warning::` or `.. attention::` block, i.e.
```
  // Allow the wasm file to include pre-compiled code on VMs which support it.
  //
  // .. warning::
  //   This should only be enable for trusted sources as the precompiled code is not verified.
```

Reply:
Done